Celigo.ServiceManager.NetSuite.REST/src/RestClient.cs

```diff
@@ -0,0 +1,203 @@
+                                                     string token,
+                                                     string tokenSecret,
+                                                     string httpMethod)
+        {
+            var oauthParams = this.GetCommonOAuthParameters();
+            this.AddQueryParamsFrom(requestUri, oauthParams);
+            
+            string basePath = requestUri.GetLeftPart(UriPartial.Path);
+            return this.GetAuthorizationHeaderValue(account, basePath, oauthParams, token, tokenSecret, httpMethod);
+        }
+        
+        protected string GetAuthorizationHeaderValue(string account,
+                                                    string baseUrlPath,
+                                                    SortedDictionary<string, string> oauthParams,
+                                                    string token,
+                                                    string tokenSecret,
+                                                    string httpMethod)
+        {
+            oauthParams.Add("oauth_token", token);
+            
+            string E(string data) => Uri.EscapeDataString(data);
+
+            var headerBuilder = new StringBuilder(300);
+            var baseStringBuilder = new StringBuilder(300);
+
+            headerBuilder.Append($"OAuth realm=\"{E(account.ToUpperInvariant().Replace('_', '-'))}\"");
+            bool isFirstParam = true;
+
+            foreach (var pair in oauthParams)
+            {
+                var encodedValue = E(pair.Value);
+
+                if (pair.Key.StartsWith("oauth_"))
+                {
+                    headerBuilder.Append(", ")
+                                .Append(pair.Key)
+                                .Append("=\"")
+                                .Append(encodedValue)
+                                .Append("\"");
+                }
+
+                if (!isFirstParam)
+                {
+                    baseStringBuilder.Append("&");
+                }
+                else
+                {
+                    isFirstParam = false;
+                }
+
+                baseStringBuilder.Append(pair.Key)
+                            .Append("=")
+                            .Append(encodedValue);
+            }
+
+            string baseString = $"{httpMethod}&{E(baseUrlPath)}&{E(baseStringBuilder.ToString())}";
+
+            return headerBuilder.Append(", oauth_signature=\"")
+                        .Append(E(this.ComputeSignature(baseString, tokenSecret)))
+                        .Append("\"")
+                        .ToString();
+        }
+
+        protected SortedDictionary<string, string> GetCommonOAuthParameters()
+        {
+            return new SortedDictionary<string, string> {
+                { "oauth_consumer_key", ConsumerKey },
+                { "oauth_nonce", this.ComputeNonce() },
+                { "oauth_signature_method", "HMAC-SHA256" },
```

Comment:
Someone can override ComputeSignature and use a different hashing algorithm, right ?

Reply:
Yup! Good catch